### ISSUE TITLE
Post made just now show up as 12 hours ago

### ISSUE THREAD

**Issue by kakadanhem** [humhub/humhub#2686]
This problem the same to https://github.com/humhub/humhub/issues/95

I have set timezone to (Asia/Phnom_Penh) at
- Administration/Setting/Basic
- User Profile/Setting

According issue above, I can use "InitSQLs" but that option is taken away for the new version (1.2) I guess as it no longer recognize as an option. It was working fine using my localhost until I upload to Hostgator.

**Comment by buddh4** [comment]
Did you check the timezone of your database?

**Comment by kakadanhem** [comment]
I suspect it probably because of mysql timezone but I'm on shared hosting, I don't think I can change that?

**Comment by Felli** [comment]
@kakadanhem what type of MySQL database are you using? (MariaDB etc.)

**Comment by kakadanhem** [comment]
Here my server info:
Server: Localhost via UNIX socket
Server type: Percona Server
Server version: 5.6.32-78.1 - Percona Server (GPL), Release 78.1, Revision 8bb53b6
Protocol version: 10

I don't think I have user with SUPER privilege, I may have to ask the hosting for that.

**Comment by Felli** [comment]
I have done some looking and found this from their community forums.
https://www.percona.com/forums/questions-discussions/mysql-and-percona-server/2873-timezone-question

**Comment by kakadanhem** [comment]
Since I'm using Shared Server, I cannot change timezone on the MySQL Server. Is there a work around? There should be a replacement for 'InitSQLs' in config/dynamic.php? or there I can edit some code to display the time correct by adding 12 hours to the time that gets from database

**Comment by Felli** [comment]
I would wait for @buddh4 to reply to that question, as I wouldn't know if there were any other options for this other than what I have given, sorry. :frowning_face:

**Comment by kakadanhem** [comment]
It's alrgiht Felli. I mean if we able to do sth like this would be nice:
```
    'db' => 
    array (
      'connectionString' => 'mysql:host=localhost;dbname=humhub_master',
      'username' => 'root',
      'password' => 'xxx',
      'initSQLs'=>array("SET time_zone = '+4:30'"),
    ),
```

**Comment by acs-ferreira** [comment]
Instead of `initSQLs`, try:

```
'on afterOpen' => function($event) {
       $event->sender->createCommand("SET time_zone = '+4:30'")->execute();
}
```

**Comment by kakadanhem** [comment]
Thank you! that work!

**Comment by acs-ferreira** [comment]
@buddh4 docs should be fixed at https://github.com/humhub/humhub/commit/b30b71889d44473ea1c7411282da4232969f419a as `initSQLs` is Yii command, not Yii2.

**Comment by kakadanhem** [comment]
After changed dynamic.php like ferreira mention it works. However, in my case, when I remove the logo in appearance setting the above part will change to
```
      'on afterOpen' => 
      Closure::__set_state(array(
      )),
```
And that make the site break: Fatal error: Call to undefined method Closure::__set_state() in ../protected/config/dynamic.php on line 12

**Comment by buddh4** [comment]
don't change the dynamic.php directly, you can overwrite values in `common.php`

**Comment by kakadanhem** [comment]
sorry, but how? can you show me what need to be done?

**Comment by buddh4** [comment]
just add the following to your `common.php` (in the same directory as `dynamic.php`:

```
<?php
return [
  'components' => [
    'db' => [
       'on afterOpen' => function($event) {
           $event->sender->createCommand("SET time_zone = '+4:30'")->execute();
        }
    ]
  ]
];
```

**Comment by kakadanhem** [comment]
Awesome, thank you!

**Comment by dosones** [comment]
It worked perfectly. I only have to set the time zone according to my time zone :-D

**Comment by 100Max** [comment]
Mine was showing as posted 2 hours ago..... even tho it was 3 seconds ago. lol!
* Fixed the windows server time that was wrong... still same thing. even after apache restart.
* I got it fixed tho:  by setting mysql database time zone. nice.

Set your mysql database timezone using php myadmin:
* Select / Load your humhub database in php myadmin.
* Click the **_SQL tab_**: and type in:    **SET GLOBAL time_zone = '-6:00';**      and then click Go.
... replace -6:00 with your correct timezone. Check the net for your correct timezone.. - or + hour.
Note: SET time_zone = 'America/Chicago';  Did Not work for me. sql error.
Using mysql 5.7+ ...apache.

* Just adding to this post to help other humub new users. 

Humhub seems to be exactly what i was looking for. and I tried a few others.
...and I know good coding when I see it. and this is it right here. humhub!
It even updates the post time without even refreshing. awesome.
HumHub rocks... It is very well done... awesome job you have done here.
Keep up the awesome work humhub.. indeed.
